RHDEVDOCS-4110: GitOps Operator to configure secure communication wit… (#62228)
openshift/openshift-docs: modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc

[source,yaml]
----
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: argocd <1>
  namespace: openshift-gitops <2>
spec:
  redis:
    autotls: openshift <3>
  ha:
    enabled: true <4>
----
+
<1> Specify a name of the Argo CD instance. In this example, the name is `argocd`.
<2> Specify a namespace where you want to run the Argo CD instance. In this example, the namespace is `openshift-gitops`.
<3> This enables the `autotls` setting and creates a TLS certificate for Redis.
<4> This enables the HA feature. If you do not want to enable HA, do not include this line or put the value as `enabled: false`.
+
[NOTE]
====
Alternatively, you can enable `autotls` on an already existing Argo CD instance by running the following command:

[source,terminal]
----
$ oc patch argocds.argoproj.io <instance-name> --type=merge -p '{"spec":{"redis":{"autotls":"openshift"}}}'
----
====
+
[NOTE]
====
The setting with HA enabled requires a cluster with at least three worker nodes.
====

.. Click *Create*.

.. Verify that the Argo CD pods are ready and running. For the Argo CD instance with HA enabled, this might take a few minutes.
+
[source,terminal]
----

> $ oc get pods

Reviewer: ```suggestion
$ oc get pods -n openshift-gitops
```

----
+
.Example output with HA disabled
[source,terminal]
----
NAME                                  READY   STATUS    RESTARTS   AGE
argocd-application-controller-0       1/1     Running   0          26s
argocd-redis-84b77d4f58-vp6zm         1/1     Running   0          37s
argocd-repo-server-5b959b57f4-znxjq   1/1     Running   0          37s
argocd-server-6b8787d686-wv9zh        1/1     Running   0          37s
----
+
.Example output with HA enabled
[source,terminal]
----
NAME                                       READY   STATUS    RESTARTS   AGE
argocd-application-controller-0            1/1     Running   0          10m
argocd-redis-ha-haproxy-669757fdb7-5xg8h   1/1     Running   0          10m
argocd-redis-ha-server-0                   2/2     Running   0          9m9s
argocd-redis-ha-server-1                   2/2     Running   0          98s
argocd-redis-ha-server-2                   2/2     Running   0          53s
argocd-repo-server-576499d46d-8hgbh        1/1     Running   0          10m
argocd-server-9486f88b7-dk2ks              1/1     Running   0          10m
----

. Verify that a secret called `argocd-operator-redis-tls` was created:
+
[source,terminal]
----

> $ oc get secrets argocd-operator-redis-tls

Reviewer: ```suggestion
$ oc get secrets argocd-operator-redis-tls -n openshift-gitops
```

.. Click *Instances* -> *Create ArgoCD*.

.. In the YAML field, replace the YAML code:
+
.Example Argo CD CR with autotls enabled
[source,yaml]
----
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: argocd <1>
  namespace: openshift-gitops <2>
spec:
  redis:
    autotls: openshift <3>
  ha:
    enabled: true <4>
----
<1> Specify a name of the Argo CD instance. In this example, the name is `argocd`.
<2> Specify a namespace where you want to run the Argo CD instance. In this example, the namespace is `openshift-gitops`.
<3> This enables the `autotls` setting and creates a TLS certificate for Redis.
<4> This enables the HA feature. If you do not want to enable HA, do not include this line or put the value as `enabled: false`.
+
[NOTE]
====
Alternatively, you can enable `autotls` on an already existing Argo CD instance by running the following command:

[source,terminal]
----
$ oc patch argocds.argoproj.io <instance-name> --type=merge -p '{"spec":{"redis":{"autotls":"openshift"}}}'
----
====
+
[NOTE]
====
The setting with HA enabled requires a cluster with at least three worker nodes.
====

.. Click *Create*.

> .. Verify that the Argo CD pods are ready and running. For the Argo CD instance with HA enabled, this might take a few minutes.

Reviewer 2: @ekristova Please follow the suggestion I gave above for a similar instance. I think it is for substep f of step 1 in the Configuring TLS for Redis with autotls disabled section for this portion of content.

----
<1> Specify a namespace where the Argo CD instance is running, for example `openshift-gitops`.
+
.Example output with HA disabled
[source,terminal]
----
NAME                                  READY   STATUS    RESTARTS   AGE
argocd-application-controller-0       1/1     Running   0          26s
argocd-redis-84b77d4f58-vp6zm         1/1     Running   0          37s
argocd-repo-server-5b959b57f4-znxjq   1/1     Running   0          37s
argocd-server-6b8787d686-wv9zh        1/1     Running   0          37s
----
+
.Example output with HA enabled
[source,terminal]
----
NAME                                       READY   STATUS    RESTARTS   AGE
argocd-application-controller-0            1/1     Running   0          10m
argocd-redis-ha-haproxy-669757fdb7-5xg8h   1/1     Running   0          10m
argocd-redis-ha-server-0                   2/2     Running   0          9m9s
argocd-redis-ha-server-1                   2/2     Running   0          98s
argocd-redis-ha-server-2                   2/2     Running   0          53s
argocd-repo-server-576499d46d-8hgbh        1/1     Running   0          10m
argocd-server-9486f88b7-dk2ks              1/1     Running   0          10m
----

> . Verify that a secret called `argocd-operator-redis-tls` was created:

Reviewer 2: ```suggestion
. Verify that a `argocd-operator-redis-tls` secret is created:
```

// Module is included in the following assemblies:
//
// * /cicd/gitops/configuring-secure-communication-with-redis.adoc

[id="gitops-configuring-tls-for-redis-with-autotls-enabled_{context}"]
= Configuring TLS for Redis with autotls enabled

[NOTE]
====
By default, the `autotls` setting is disabled.
====

> 

Reviewer 2: @ekristova You need to add an introductory sentence or two for this procedure module file. See here for details on the different parts of a procedure module: https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs

====
+
[NOTE]
====
The HA-enabled TLS configuration requires a cluster with at least three worker nodes.
====
+
.Example output with HA enabled
[source,terminal]
----
NAME                                       READY   STATUS    RESTARTS   AGE
argocd-application-controller-0            1/1     Running   0          10m
argocd-redis-ha-haproxy-669757fdb7-5xg8h   1/1     Running   0          10m
argocd-redis-ha-server-0                   2/2     Running   0          9m9s
argocd-redis-ha-server-1                   2/2     Running   0          98s
argocd-redis-ha-server-2                   2/2     Running   0          53s
argocd-repo-server-576499d46d-8hgbh        1/1     Running   0          10m
argocd-server-9486f88b7-dk2ks              1/1     Running   0          10m
----

. Verify that a `argocd-operator-redis-tls` secret is created:
+
[source,terminal]
----
$ oc get secrets argocd-operator-redis-tls -n <namespace> <1>
----
<1> Specify a namespace where the Argo CD instance is running, for example `openshift-gitops`.
+
.Example output 
[source,terminal]
----
NAME                        TYPE                DATA   AGE
argocd-operator-redis-tls   kubernetes.io/tls   2      30s
----
+

> The secret should be the `kubernetes.io/tls` type and should have a size of `2`.

Reviewer 2: ```suggestion
The secret must be the `kubernetes.io/tls` type and a size of `2`.
```
@ekristova Per the ISG: "Do not use should to refer to an action that must be performed." https://www.ibm.com/docs/en/ibm-style?topic=word-usage#s

@reginapizza and @varshab1210 PTAL and confirm that my suggestion does not change the technical accuracy. Thanks!

+
[NOTE]
====
The HA-enabled TLS configuration requires a cluster with at least three worker nodes.
====
+
.Example output with HA enabled
[source,terminal]
----
NAME                                       READY   STATUS    RESTARTS   AGE
argocd-application-controller-0            1/1     Running   0          10m
argocd-redis-ha-haproxy-669757fdb7-5xg8h   1/1     Running   0          10m
argocd-redis-ha-server-0                   2/2     Running   0          9m9s
argocd-redis-ha-server-1                   2/2     Running   0          98s
argocd-redis-ha-server-2                   2/2     Running   0          53s
argocd-repo-server-576499d46d-8hgbh        1/1     Running   0          10m
argocd-server-9486f88b7-dk2ks              1/1     Running   0          10m
----

. Verify that the `argocd-operator-redis-tls` secret is created:
+
[source,terminal]
----
$ oc get secrets argocd-operator-redis-tls -n <namespace> <1>
----
<1> Specify a namespace where the Argo CD instance is running, for example `openshift-gitops`.
+
.Example output 
[source,terminal]
----
NAME                        TYPE                DATA   AGE
argocd-operator-redis-tls   kubernetes.io/tls   2      30s
----
+

> Ensure that the secret is the `kubernetes.io/tls` type and a size of `2`.

Reviewer 2: ```suggestion
The secret must be of the `kubernetes.io/tls` type and a size of `2`.
```
